python/lsst/drp/tasks/update_visit_summary.py

```diff
@@ -0,0 +1,762 @@
+    """List of tuples of catalogs and the tracts they correspond to
+    (`list` [`tuple` [`lsst.skymap.TractInfo`,
+        `lsst.afw.table.ExposureCatalog`]]).
+    """
+
+    @classmethod
+    def load(
+        cls,
+        butler: ButlerQuantumContext | Butler,
+        sky_map: BaseSkyMap,
+        refs: Iterable[DatasetRef],
+    ) -> PerTractInput:
+        """Load and wrap input catalogs.
+
+        Parameters
+        ----------
+        butler : `lsst.pipe.base.ButlerQuantumContext`
+            Butler proxy used in `~lsst.pipe.base.PipelineTask.runQuantum`.
+        sky_map : `lsst.skymap.BaseSkyMap`
+            Definition of tracts and patches.
+        refs : `~collections.abc.Iterable` [`lsst.daf.butler.DatasetRef`]
+            References to the catalog datasets to load.
+
+        Returns
+        -------
+        wrapper : `PerTractInput`
+            Wrapper object for the loaded catalogs.
+        """
+        catalogs_by_tract = []
+        for ref in refs:
+            tract_id = ref.dataId["tract"]
+            tract_info = sky_map[tract_id]
+            catalogs_by_tract.append(
+                (
+                    tract_info,
+                    butler.get(ref),
+                )
+            )
+        return cls(catalogs_by_tract)
+
+    def best_for_detector(
+        self,
+        detector_id: int,
+        center: SpherePoint | None = None,
+        bbox: Box2I | None = None,
+    ) -> tuple[int, ExposureRecord | None]:
+        # Docstring inherited.
+        best_result: tuple[int, ExposureRecord | None] = (-1, None)
+        best_distance: Angle = float("inf") * degrees
+        for tract_info, catalog in self.catalogs_by_tract:
+            record = catalog.find(detector_id)
+            if record is None:
+                continue
+            if center is None:
+                center_for_record = compute_center_for_detector_record(
+                    record, bbox=bbox
+                )
+                if center_for_record is None:
+                    continue
+            else:
+                center_for_record = center
+            center_distance = tract_info.ctr_coord.separation(center_for_record)
+            if best_distance > center_distance:
+                best_result = (tract_info.tract_id, record)
+                best_distance = center_distance
+        return best_result
+
+
+@dataclasses.dataclass
+class GlobalInput(PossiblyMultipleInput):
+    """Wrapper class for input `~lsst.afw.table.ExposureCatalog` datasets
+    that are not per-tract.
+    """
+
+    catalog: ExposureCatalog
+    """Loaded per-visit catalog dataset (`lsst.afw.table.ExposureCatalog`).
+    """
+
+    def best_for_detector(
+        self,
+        detector_id: int,
+        center: SpherePoint | None = None,
+        bbox: Box2I | None = None,
+    ) -> tuple[int, ExposureRecord | None]:
+        # Docstring inherited.
+        return -1, self.catalog.find(detector_id)
```

Comment:
I have a memory of writing this comment before, but I'm not seeing it now, so just in case (sorry if it's a dup!)

Does it matter that the tuple-making parentheses are not included in the return?

Reply:
```>>> x = 1, 2 ,3
>>> type(x)
<class 'tuple'>```